### DISCUSSION TITLE
Asset file localization

### DISCUSSION THREAD

**Discussion by sidedwards** [craftcms/cms#10233] [Ideas] [answered]
## 📜 Context
It is currently possible to localize Asset fields and store values on a per-site basis. However, it is not possible to localize Asset files, which means users must create and reference separate Assets if they want to localize their files. It would be great to be able to store Asset files on a per-site basis as well.

## 📚 Resources

<img width="1242" alt="image" src="https://user-images.githubusercontent.com/11489608/145433914-6d4f9ec8-3792-4e3f-a72e-b9acd018d6a4.png">

Asset file localization within Contentful CMS
<img width="1108" alt="image" src="https://user-images.githubusercontent.com/11489608/145434907-5eada392-c950-414f-a36f-129172af7b48.png">

**Comment by brandonkelly** [comment, marked as answer]
You can sortof pull this off with asset relations:

1. Go to **Settings** → **Fields** and create a new Assets field with the following settings:
    - **Name**: Localized Images
    - **Handle**: `localizedImages`
    - **Field Type**: Assets
    - ✅  **Restrict assets to a single folder**
    - **Asset Location**: [your volume], `localized/{site.language}`
    - ✅  **Restrict allowed file types** → **Image**
    - ✅  **Allow uploading directly to the field**
    - **Limit**: 1
    - **Advanced**
        - ✅  **Manage relations on a per-site basis**
2. Go to your **Settings** → **Volumes** → [your volume] and add the new Localized Images field to its field layout

Then when editing images for the volume, you can optionally upload site-specific assets to them via the Localized Images field.

In your templates, you can then check if the current asset has a localized image, and output that instead:

```twig
{% set locImage = image.localizedImages.one() %}
{{ (locImage ?? image).getImg() }}
```

**Reply by sidedwards** [reply]
@brandonkelly, thank you for the detailed response; I think a localized relational asset field could work in most scenarios.

Do you think it would ever make sense to allow native localization for asset files v.s a custom field? We have a few clients interested in localizing asset files. While we could likely support custom relational asset fields with localized volumes, it would be great to leverage native functionality if possible.

**Reply by brandonkelly** [reply]
It’s come up before and I’m not completely opposed to the idea, but it’s a pretty drastic shift from how Assets work right now, where there is a one-to-one map between the elements and files on the volume.
